src/js/misc.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  Event,
+} from '@sentry/types';
+
+/**
+ * Recognizes an Event based on 'exception' property.
+ */
+export function isEvent(event: unknown): event is Event {
+  return typeof event === 'object' && event !== null && 'exception' in event;
+}
+
+/**
+ * Extracts the hard crash information from the event exceptions.
+ * No exceptions or undefined handled are not hard crashes.
+ */
+export function isHardCrash(event: Event): boolean {
+  let isHandled = true;
+
+  for (const exception of event.exception?.values ?? []) {
+    if (!(exception.mechanism?.handled !== false)) {
+      isHandled = false;
```

Comment:
You can return `false` early here.

Reply:
Thanks, yes and safe time going thru the rest of the exceptions.